url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/model/TimeSpacingPrefs.java

```diff
@@ -0,0 +1,100 @@
+package coursescheduleapp.model;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class TimeSpacingPrefs implements Preference {
+
+	private double score;
+	private String courseSpacing;
+	
+	public TimeSpacingPrefs()
+	{
+		courseSpacing = CourseSpacing.NOPREF.toString();
+		score = .1;
+	}
+	public TimeSpacingPrefs(String s)
+	{
+		courseSpacing = s;
+		score = .1;
+	}
+	
+	public void setSpacing(String s)
+	{
+		courseSpacing  = s;
+	}
+	
+	
+	@Override
+	public double getFitness(Schedule individual) {
+		
+		if (courseSpacing.equals(CourseSpacing.NOPREF.toString()))
+		{
+			return 0;
+		}
+		
+		TimeDayCollection sched = new TimeDayCollection();
+		Iterator<Course> it = individual.getCourses().iterator();
+
+		// Iterate over each course to create a time table for the days and times
+		// of the courses.
+		while(it.hasNext()) 
+		{
+			Course curr = it.next();
+			// I expect that each course has an ArrayList of Day enums
+			// and a TimePeriod object
+			if (curr != null)
+			{
+				boolean wasAdded = sched.addTimeCollection((TimePeriod)curr.getValue(CourseData.TIME.toString()),(ArrayList<Day>)curr.getValue(CourseData.DAY.toString()));
+				// If wasAdded is false then two courses occur during or right after the other
+				// like if one class starts at 10am and the other class ends at 10am
+				if (wasAdded == false)
+				{
+					return 0;
+				}
+			}
+		}
+		return getFitness(sched);
+	}
+	private double getFitness(TimeDayCollection sched) {
+		double fitness = 0;
+		if (courseSpacing != CourseSpacing.NOPREF.toString())
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.